core/trino-main/src/test/java/io/trino/sql/analyzer/TestAnalyzer.java

```diff
@@ -6622,6 +6623,33 @@ public void testTableFunctionAliasing()
                 .hasMessage("line 1:23: Column 'table_alias.a' cannot be resolved");
     }
 
+    @Test
+    public void testTableFunctionRequiredColumns()
+    {
+        // the function required_column_function specifies columns "A" (capitalized) and "b" from table argument "INPUT" as required.
+        analyze("""
+                SELECT * FROM TABLE(system.required_columns_function(
+                    input => TABLE(t1)))
+                """);
+
+        analyze("""
+                SELECT * FROM TABLE(system.required_columns_function(
+                    input => TABLE(SELECT 1, 2, 3) t(x, b, a)))
+                """);
+
+        analyze("""
+                SELECT * FROM TABLE(system.required_columns_function(
+                    input => TABLE(SELECT 1, 2) t(A, B)))
+                """);
+
+        assertFails("""
+                SELECT * FROM TABLE(system.required_columns_function(
+                    input => TABLE(SELECT 1, 2) t(A, C)))
+                """)
+                .hasErrorCode(COLUMN_NOT_FOUND)
+                .hasMessage("Column 'b' cannot be resolved");
```

Comment:
This error is confusing. There's nothing actionable for the user that submits such a query. The fact that the function requires `b` is an implementation error. The error should indicate so.

Reply:
Changed.